lua/ui/game/reclaim.lua

```diff
@@ -1,497 +1,494 @@
+local MathMax = math.max
+local MathMin = math.min
+local TableGetn = table.getn
+local TableSort = table.sort
 
----@class UIReclaimDataCombined
+---@class UserReclaimData
 ---@field mass number
 ---@field position Vector
 ---@field count number
 ---@field max number
 
+---@class WorldLabelGroup : Group
+---@field view WorldView
+---@field isMoving boolean
+---@field prevPos Vector
+
+
 ---@type number
-local HEIGHT_RATIO = 0.012
+local HeightRatio = 0.012
 
---- Reclaim is no longer combined once this threshold is met, the value (150) is the same
---- camera distance that allows for the reclaim command to work. Guarantees that the
---- labels represent where you can reclaim.
+--- Reclaim is no longer combined once this threshold is met, the value (150) is the same camera
+--- distance that allows for the reclaim command to work. Guarantees that the  labels represent
+--- where you can reclaim.
 ---@type number
-local ZOOM_THRESHOLD = 150
+local ZoomThreshold = 150
 
 --- TODO: remove the options
 ---@type number
 local MaxLabels = options.maximum_reclaim_count or 1000
-
 --- TODO: remove the options
 ---@type number
 local MinAmount = options.minimum_reclaim_amount or 10
 
----@type table<EntityId, UIReclaimDataPoint>
+---@type table<EntityId, PropSyncData>
 local Reclaim = {}
-
----@type table<EntityId, UIReclaimDataPoint>
+---@type table<EntityId, PropSyncData>
 local OutsidePlayableAreaReclaim = {}
 
----@type UIReclaimDataCombined[]
-local reclaimDataPool = {}
-
----@type number
-local totalReclaimData = 0
+---@type UserReclaimData[], number, number
+local DataPool, DataPoolSize, LastDataSize = {}, 0, 0
 
----@type WorldLabel[]
-local LabelPool = {}
+---@type WorldLabel[], number, number
+local LabelPool, LabelPoolSize, LastLabelCount = {}, 0, 0
 
----@type number
-local OldZoom
+---@type UserRect
+local PlayableArea
 
----@type Vector
-local OldPosition
+---@type number, number
+local MapWidth, MapHeight = 0, 0
 
----@type boolean
+local CommandGraphActive = false
 local ReclaimChanged = true
-local PlayableArea
 
----@class WorldLabel : Group
----@field parent WorldView
----@field position Vector
----@field mass Bitmap
----@field text Text
-local WorldLabel = Class(Group) {
-    __init = function(self, parent, position)
-        Group.__init(self, parent)
-        self.parent = parent
-        self.position = position
-        self._project = true
-
-        self.Top:Set(0)
-        self.Left:Set(0)
-        LayoutHelpers.SetDimensions(self, 25, 25)
 
-        self.mass = Bitmap(self)
-        self.mass:SetTexture(UIUtil.UIFile('/game/build-ui/icon-mass_bmp.dds'))
-        LayoutHelpers.AtLeftIn(self.mass, self)
-        LayoutHelpers.AtVerticalCenterIn(self.mass, self)
-        LayoutHelpers.SetDimensions(self.mass, 14, 14)
+---@class WorldLabel : Label
+---@field icon Bitmap
+---@field text Text
+---
+---@field parent WorldLabelGroup
+---@field position Vector
+---@field reproject boolean
+local WorldLabel = Class(Label) {
```

Comment:
feels like we already have the changes here pinned down, no need to include this